lang/go/go.py

```diff
@@ -0,0 +1,213 @@
+def go_types(m) -> str:
+    "Returns a string"
+    return m.go_types
+
+
+@mod.capture(rule="{self.go_pointers}")
+def go_pointers(m) -> str:
+    "Returns a string"
+    return m.go_pointers
+
+
+@ctx.action_class("user")
+class UserActions:
+    def code_operator_lambda():
+        actions.insert(" -> ")
+
+    def code_operator_subscript():
+        actions.user.insert_between("[", "]")
+
+    def code_operator_assignment():
+        actions.insert(" = ")
+
+    def code_operator_subtraction():
+        actions.insert(" - ")
+
+    def code_operator_subtraction_assignment():
+        actions.insert(" -= ")
+
+    def code_operator_addition():
+        actions.insert(" + ")
+
+    def code_operator_addition_assignment():
+        actions.insert(" += ")
+
+    def code_operator_multiplication():
+        actions.insert(" * ")
+
+    def code_operator_multiplication_assignment():
+        actions.insert(" *= ")
+
+    def code_operator_exponent():
+        actions.insert(" ^ ")
+
+    def code_operator_division():
+        actions.insert(" / ")
+
+    def code_operator_division_assignment():
+        actions.insert(" /= ")
+
+    def code_operator_modulo():
+        actions.insert(" % ")
+
+    def code_operator_modulo_assignment():
+        actions.insert(" %= ")
+
+    def code_operator_equal():
+        actions.insert(" == ")
+
+    def code_operator_not_equal():
+        actions.insert(" != ")
+
+    def code_operator_greater_than():
+        actions.insert(" > ")
+
+    def code_operator_greater_than_or_equal_to():
+        actions.insert(" >= ")
+
+    def code_operator_less_than():
+        actions.insert(" < ")
+
+    def code_operator_less_than_or_equal_to():
+        actions.insert(" <= ")
+
+    def code_operator_and():
+        actions.insert(" && ")
+
+    def code_operator_or():
+        actions.insert(" || ")
+
+    def code_operator_bitwise_and():
+        actions.insert(" & ")
+
+    def code_operator_bitwise_and_assignment():
+        actions.insert(" &= ")
+
+    def code_operator_increment():
+        actions.insert("++")
+
+    def code_operator_bitwise_or():
+        actions.insert(" | ")
+
+    def code_operator_bitwise_exclusive_or():
+        actions.insert(" ^ ")
+
+    def code_operator_bitwise_left_shift():
+        actions.insert(" << ")
+
+    def code_operator_bitwise_left_shift_assignment():
+        actions.insert(" <<= ")
+
+    def code_operator_bitwise_right_shift():
+        actions.insert(" >> ")
+
+    def code_operator_bitwise_right_shift_assignment():
+        actions.insert(" >>= ")
+
+    def code_self():
+        actions.insert("this")
+
+    def code_operator_object_accessor():
+        actions.insert(".")
+
+    def code_insert_null():
+        actions.insert("nil")
+
+    def code_insert_is_null():
+        actions.insert(" == nil")
+
+    def code_insert_is_not_null():
+        actions.insert(" != nil")
+
+    def code_state_if():
+        actions.user.insert_between("if ", " ")
```

Comment:
What is the `" "` doing here?

Reply:
The same action inserts the parens in JS and other C-like languages, so no.

Reply:
Go does not require them.

Reply:
I'm suggesting adding the opening bracket of the if block, not the opening paren of the if considering. I'm not sure if that was clear.